test/integration/query-interface/schemas.test.ts

```diff
@@ -0,0 +1,47 @@
+import { expect } from 'chai';
+import { afterEach } from 'mocha';
+import { spy } from 'sinon';
+import type { CreateSchemaQueryOptions } from '@sequelize/core/_non-semver-use-at-your-own-risk_/dialects/abstract/query-generator';
+import { sequelize } from '../support';
+
+const queryInterface = sequelize.queryInterface;
+
+const testSchema = 'testSchema';
+
+if (sequelize.dialect.supports.schemas) {
+  describe('QueryInterface#{create}Schema', async () => {
+
+    // Clear the test schema if it's been added
+    afterEach(async () => {
+      const schemaNames: string[] = await queryInterface.showAllSchemas();
+      if (schemaNames.includes(testSchema)) {
+        await queryInterface.dropSchema(testSchema);
+      }
+    });
+
+    it('should create a schema', async () => {
+      const preCreationSchemas = await queryInterface.showAllSchemas();
+      expect(preCreationSchemas).to.not.include(testSchema, 'testSchema existed before tests ran');
+
+      await queryInterface.createSchema(testSchema);
+      const postCreationSchemas: string[] = await queryInterface.showAllSchemas();
```

Comment:
```suggestion
      const postCreationSchemas = await queryInterface.showAllSchemas();
```
TypeScript should already know that the return of `showAllSchemas` is `string[]`, right?

Reply:
Yeah, I just like explicit typing. Will remove